pyfstat/mcmc_based_searches.py

```diff
@@ -1731,15 +1720,15 @@ def _check_old_data_is_okay_to_use(self):
 
     def get_savetxt_fmt(self):
         fmt = helper_functions.get_doppler_params_output_format(self.theta_keys)
-        fmt += ["%.9g"]  # for detection statistic
+        fmt["twoF"] = "%.9g"
         return fmt
 
     def export_samples_to_disk(self):
         self.samples_file = os.path.join(self.outdir, self.label + "_samples.dat")
         logging.info("Exporting samples to {}".format(self.samples_file))
         header = "\n".join(self.output_file_header)
         header += "\n" + " ".join(self.theta_keys) + " twoF"
-        outfmt = self.get_savetxt_fmt()
+        outfmt = list(self.get_savetxt_fmt().values())
```

Comment:
Ok!

Reply:
Actually, not ok! This will result in `outfmt` in the order the dict was filled, which does not need to be the same as the order of samples columns. This currently causes a bug at least for `MCMCGlitchSearch`.

Reply:
I think the `MCMCGlitchSearch` glitch output was already buggy on master, see #219, though this PR initially made it worse. ;). Should now be fixed in 662d47217d97ef528663f121f0e02f432b0e3fad but I still need to double-check if similar issues appear in the grid classes.

Reply:
This looks like on of those "Here's yet another way of writing data down!" which #191 should take care of. I'll mark this.